plugins/pluginManager.js

```diff
@@ -1494,22 +1494,36 @@ var pluginManager = function pluginManager() {
         var toReturn = {};
 
         for (let i in ob) {
-            if (!ob.hasOwnProperty(i)) {
+            if (!Object.prototype.hasOwnProperty.call(ob, i)) {
                 continue;
             }
 
             if ((typeof ob[i]) === 'object' && ob[i] !== null) {
                 var flatObject = flattenObject(ob[i]);
                 for (let x in flatObject) {
-                    if (!flatObject.hasOwnProperty(x)) {
+                    if (!Object.prototype.hasOwnProperty.call(flatObject, x)) {
                         continue;
                     }
 
                     toReturn[prefix + i + '.' + x] = flatObject[x];
                 }
             }
             else {
-                ob[i] = (!isNaN(ob[i]) && ob[i] > 2147483647) ? 2147483647 : ob[i];
+                const configsKey = Object.keys(configs);
+                const defaultConfigsKey = Object.keys(defaultConfigs);
+                const objectConfigs = {...configs, ...defaultConfigs};
+                const arrConfigsKey = [...configsKey, ...defaultConfigsKey];
+                for (let configKey of arrConfigsKey) {
+                    const objectKey = Object.keys(objectConfigs[configKey]);
+                    objectKey.map(key=>{
```

Comment:
Here `forEach` maybe is better suitable than `map`

Reply:
In overall, this is really not optimal, we are traversing all the nodes of object, and inside each traverse, we do an iteration on all possible keys. It creates complexity exponentially.
I think you can just get the needed key value from prefix variable (it's all upper keys with dot as divider) and use that to check defaultConfigs value type in constant time